tests/integration/high_availability/test_replication.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+
+import lightkube
+import pytest
+import yaml
+from helpers import get_cluster_status, get_primary_unit
+from lightkube.resources.core_v1 import Pod
+from pytest_operator.plugin import OpsTest
+from tenacity import RetryError, Retrying, stop_after_delay, wait_fixed
+
+from tests.integration.high_availability.high_availability_helpers import (
+    get_max_written_value_in_database,
+)
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+MYSQL_APP_NAME = METADATA["name"]
+APPLICATION_APP_NAME = "application"
+CLUSTER_NAME = "test_cluster"
+TIMEOUT = 15 * 60
+
+
+@pytest.mark.order(1)
+@pytest.mark.abort_on_fail
+@pytest.mark.replication_tests
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build, deploy and relate the mysql and test applications."""
+    mysql_charm = await ops_test.build_charm(".")
+    application_charm = await ops_test.build_charm(
+        "./tests/integration/high_availability/application_charm/"
+    )
+
+    mysql_config = {"cluster-name": CLUSTER_NAME}
+    mysql_resources = {"mysql-image": METADATA["resources"]["mysql-image"]["upstream-source"]}
+
+    async with ops_test.fast_forward():
+        await ops_test.model.deploy(
+            mysql_charm,
+            application_name=MYSQL_APP_NAME,
+            config=mysql_config,
+            resources=mysql_resources,
+            num_units=3,
+        )
+
+        await ops_test.model.deploy(
+            application_charm,
+            application_name=APPLICATION_APP_NAME,
+            num_units=1,
+        )
+
+        await ops_test.model.relate(
+            f"{MYSQL_APP_NAME}:database", f"{APPLICATION_APP_NAME}:database"
+        )
+
+        await ops_test.model.wait_for_idle(
+            apps=[MYSQL_APP_NAME, APPLICATION_APP_NAME],
+            status="active",
+            raise_on_blocked=True,
+            timeout=TIMEOUT,
+        )
+
+        assert len(ops_test.model.applications[MYSQL_APP_NAME].units) == 3
+        assert len(ops_test.model.applications[APPLICATION_APP_NAME].units) == 1
+
+
+@pytest.mark.order(2)
+@pytest.mark.abort_on_fail
+@pytest.mark.replication_tests
+async def test_kill_primary_check_reelection(ops_test: OpsTest) -> None:
+    """Test to kill the primary under load and ensure re-election of primary."""
```

Comment:
We should use [this function](https://github.com/canonical/mysql-operator/blob/fd6dbb6d75e29fcdbbb433510e4cf2b47b6df54c/tests/integration/test_ha.py#L132) to retrieve the `MYSQL_APP_NAME` - This enables running the tests against an existing cluster.

Reply:
addressed in 0986dce467170ce6ac07013fda91adecffcf63ab - created a helper for set up where the above-like function is called for both mysql and the application